p2p/p2p.go

```diff
@@ -407,10 +407,11 @@ func (s *Service) persistPeers() error {
 	}
 
 	store := s.host.Peerstore()
-	peers := utils.Filter(store.Peers(), func(peerID peer.ID) bool {
-		return peerID != s.host.ID()
-	})
+	peers := store.Peers()
 	for _, peerID := range peers {
+		if peerID == s.host.ID() {
+			continue
+		}
```

Comment:
There is `s.log.Infow("Stored peers", "num", len(peers))` below in this case it won't log correct value

Reply:
Yes, you're right. I think it's ok to have 2 iterations in this case.